controller/lib/hal/hal_stm32.cpp

```diff
@@ -0,0 +1,1008 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+This file implements the HAL (Hardware Abstraction Layer) for the
+STM32L452 processor used on the controller.  Details of the processor's
+peripherals can be found in the reference manual for that processor:
+   https://www.st.com/resource/en/reference_manual/dm00151940-stm32l41xxx42xxx43xxx44xxx45xxx46xxx-advanced-armbased-32bit-mcus-stmicroelectronics.pdf
+
+Details specific to the ARM processor used in this chip can be found in
+the programmer's manual for the processor available here:
+   https://www.st.com/resource/en/programming_manual/dm00046982-stm32-cortexm4-mcus-and-mpus-programming-manual-stmicroelectronics.pdf
+
+*/
+
+#if defined(BARE_STM32)
+
+#include "hal.h"
+#include "hal_stm32.h"
+
+#define SYSTEM_STACK_SIZE       500
+
+// This is the main stack used in our system.
+__attribute__((aligned (8))) uint32_t systemStack[ SYSTEM_STACK_SIZE ];
```

Comment:
OK, that helps.  I was getting really annoyed at the clang-format pre-commit stuff last night because it was so inconsistent in what it would change.  At least now after running the whole files through the formatter they have a consistent style.